source/shared/cpp/ObjectModel/MarkDownHtmlGenerator.cpp

```diff
@@ -0,0 +1,198 @@
+#include "MarkDownHtmlGenerator.h"
+
+using namespace AdaptiveCards;
+
+std::string MarkDownStringHtmlGenerator::GenerateHtmlString()
+{
+    if (m_isHead) 
+    {              
+        m_token = (m_isHeadOfNewBlock)? "\n<p>" + m_token :"<p>" + m_token;
```

Comment:
>m_token = (m_isHeadOfNewBlock)? "\n<p>" + m_token :"<p>" + m_token; [](start = 8, length = 67)

Can you explain which scenarios need a newline and which don't?  Currently i don't see anywhere in the code where m_isHeadOfNewBlock is set to true.

Reply:
if there are going to be different types of blocks with different tags such as <p> and <ul>, then we append '\n'
example of usage is at MarkDownParserResult.cpp at line 35

---
In reply to: [157635494](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157635494) [](ancestors = 157635494)

Reply:
I see.  Is that required by html?

---
In reply to: [157635911](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157635911) [](ancestors = 157635911,157635494)

Reply:
it may not needed; it's probably needed to make it more readable to humans; i'm based it on Common Mark's markdown simulator.


---
In reply to: [157636215](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157636215) [](ancestors = 157636215,157635911,157635494)

Reply:
Since this html isn't really intended to be read by humans anyway, if we don't need it i would recommend removing this in order to simplify the code.  That lets you get rid of this headofnewblock member, and i think the IsList function as well.

---
In reply to: [157636474](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157636474) [](ancestors = 157636474,157636215,157635911,157635494)

Reply:
sounds good to me. isList is needed, because we want to insert different block tags <p> vs <li>

---
In reply to: [157636850](https://github.com/Microsoft/AdaptiveCards/pull/1045#discussion_r157636850) [](ancestors = 157636850,157636474,157636215,157635911,157635494)